lib/r10k/api.rb

```diff
@@ -97,6 +97,20 @@ def parse_deployed_env(path, moduledir="modules")
     # @param path [String] Path on disk to search for Puppet environments.
     # @return [Array<Hash>] An array of hashmaps, each representing the actual state of a single environment found in environmentdir.
     def parse_environmentdir(path)
+      deployed_env_states = []
+
+      if path
+        dir_entries = Dir.glob(File.join(path, '*'))
+
+        if !dir_entries.empty?
```

Comment:
We don't need to check to see if the list is empty; calling `Array#select` on an empty list will execute zero times; same with `Array#each`. So all in all, we can omit this check and the code will still behave correctly.